packages/tables/frontend/sirius-components-tables/src/TableContent.tsx

```diff
@@ -65,11 +75,24 @@ export const TableContent = ({ editingContextId, representationId, table, readOn
 
   const [editedLines, setEditedLines] = useState<GQLLine[]>(table.lines);
 
+  const [state, setState] = useState<ToolbarSettingsActioButtonState>({
+    contextMenuAnchorElement: null,
+  });
+
+  const onClickSettingButton: React.MouseEventHandler<HTMLButtonElement> = (event) =>
+    setState((prevState) => ({ ...prevState, contextMenuAnchorElement: event.currentTarget }));
+
+  const onCloseSettingContextMenu = () => setState((prevState) => ({ ...prevState, contextMenuAnchorElement: null }));
```

Comment:
All this can be extracted in a dedicated component, it does not have to be here. I'll move it to make responsibilities more clear